src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,183 @@
+use std::iter;
+use std::io::{Result, Write};
+
+use env::Env;
+use file_saver::save_to_file;
+use library::{self, Library, ToStripperType};
+use library::Type as LType;
+
+use stripper_interface::{FILE, FILE_COMMENT, MOD_COMMENT, TypeStruct};
+
+const TAB_SIZE : usize = 4;
+
+pub fn generate(env: &Env) {
+    let path =  env.config.target_path.join("doc.cmts");
+    println!("Generating documentation {:?}", path);
+    save_to_file(&path, env.config.make_backup,
+        |w| generate_doc(w, &env.library));
+}
+
+fn generate_doc(w: &mut Write, lib: &Library) -> Result<()> {
+    for i in 0..lib.namespaces.len() {
+        let mut output = String::new();
+        let namespace = &lib.namespaces[i];
+
+        if let Some(ref namespace_doc) = namespace.doc {
+            for line in namespace_doc.split("\n") {
+                output.push_str(&format!("{}{}\n", FILE_COMMENT, line));
+            }
+        }
+        try!(create_sub_docs(&mut output, &namespace.constants, None));
+        for i in 0..namespace.types.len() {
+            if let Some(ref n_ty) = namespace.types[i] {
+                try!(handle_type(&mut output, &n_ty, None, lib));
+            }
+        }
+        if output.len() > 0 {
+            try!(writeln!(w, "{}src/auto/{}.rs", FILE, namespace.name));
+            try!(writeln!(w, "{}", &output));
+        }
+    }
+    Ok(())
+}
+
+fn handle_type(w: &mut String, ty: &library::Type, parent: Option<Box<TypeStruct>>, lib: &Library) -> Result<()> {
+    match *ty {
+        LType::Alias(ref a) => create_sub_doc(w, a, parent),
+        LType::Enumeration(ref e) => create_enum_doc(w, &e, parent),
+        LType::Function(ref f) => create_fn_doc(w, &f, parent),
+        LType::Interface(ref i) => create_trait_doc(w, &i, parent),
+        LType::Class(ref c) => create_class_doc(w, &c, parent, lib),
+        _ => Ok(()),
+    }
+}
+
+fn create_class_doc(w: &mut String, class: &library::Class, parent: Option<Box<TypeStruct>>, lib: &Library) -> Result<()> {
+    let mut ty = class.convert();
+    ty.parent = parent;
+    let tabs : String = iter::repeat("    ").take(compute_indent(&ty.parent) * TAB_SIZE).collect::<Vec<&str>>().concat();
+
+    if let Some(ref class_doc) = class.doc {
+        w.push_str(&format!("{}{}\n{}\n",
+                            MOD_COMMENT,
+                            ty,
+                            format!("{}/// {}\n",
+                                    tabs,
+                                    class_doc.split("\n").collect::<Vec<&str>>().join(&format!("\n{}/// ", tabs)))));
+    }
+    for child_id in class.children.iter() {
+        if let &LType::Class(ref child) = lib.type_(child_id.clone()) {
+            try!(create_class_doc(w, &child, Some(Box::new(ty.clone())), lib));
```

Comment:
This feels like it will generate a lot of duplicate entries. Won't `handle_type` call `create_class_doc` on all children anyway?

No `create_fn_doc`?